extensions/indexes/lucene/src/main/java/org/exist/indexing/lucene/LuceneIndexWorker.java

```diff
@@ -565,6 +565,23 @@ public void compute(DirectoryTaxonomyReader reader, FacetsConfig config, FacetsC
         }
     }
 
+    /**
+     * Calls {@link LuceneUtil#extractTerms(Query, Map, IndexReader, boolean)}  to extract
+     * the terms which would be matched by the given query.
+     *
+     * @param query to extract terms for
+     * @return the map returned by {@link LuceneUtil#extractTerms(Query, Map, IndexReader, boolean)}
+     * @throws IOException in case of Lucene IO error
+     */
+    public Map<Object, Query> getTerms(final Query query) throws IOException {
+        final Map<Object, Query> termMap = new TreeMap<>();
+        index.withReader(reader -> {
+            LuceneUtil.extractTerms(query, termMap, reader, false);
+            return termMap;
+        });
+        return termMap;
```

Comment:
so this effectively only returns an empty TreeMap no?

Reply:
I am not that familiar with the `->` notation in Java. It is likely that termMap is mutated in `LuceneUtil.extractTerms` and that resulting map is then returned.

Reply:
Indeed, termMap may be mutated, but the formulation is pretty confusing. Perhaps instead consider:

```java
        return index.withReader(reader -> {
            final Map<Object, Query> termMap = new TreeMap<>();
            LuceneUtil.extractTerms(query, termMap, reader, false);
            return termMap;
        });
```

Reply:
much better @adamretter